Core/NoOpRunner.Core/Builders/GameStateBuilder.cs

```diff
@@ -0,0 +1,23 @@
+﻿using NoOpRunner.Core.Configurators;
+using NoOpRunner.Core.Interfaces;
+using System;
+
+namespace NoOpRunner.Core.Builders
+{
+    public class GameStateBuilder
+    {
+        private IGameStateConfigurator configurator;
+
+        public IGameStateConfigurator Configure()
+        {
+            return configurator = new GameStateConfigurator();
+        }
+
+        public GameState Build()
+        {
+            if (configurator == null) throw new Exception("Configurator uninitialized");
```

Comment:
When will configurator be null, if you init it in ctor and it's a private member?

Reply:
It's not initialized in the constructor, but in the Configure() method